docs/book/checkout.rst

```diff
@@ -0,0 +1,259 @@
+
+Checkout State Machine
+----------------------
+
+The Order Checkout state machine has 5 states available: ``cart``, ``addressed``, ``shipping_selected``, ``payment_selected``, ``completed``
+and a set of defined transitions between them - which are saved as the **checkoutState** of the **Order**.
+
+Besides the steps of checkout, each of them can be done once more. For instance if the Customer changes their mind
+and after selecting payment they want to change the shipping address they have already specified, they can of course go back and readdress it.
+
+The transitions on the order checkout state machine are:
+
+.. code-block:: yaml
+
+   transitions:
+      address:
+          from: [cart]
+          to: addressed
+      readdress:
+          from: [payment_selected, shipping_selected, addressed]
+          to: cart
+      select_shipping:
+          from: [addressed]
+          to: shipping_selected
+      reselect_shipping:
+          from: [payment_selected, shipping_selected]
+          to: addressed
+      select_payment:
+          from: [shipping_selected]
+          to: payment_selected
+      reselect_payment:
+          from: [payment_selected]
+          to: shipping_selected
+      complete:
+          from: [payment_selected]
+          to: completed
+
+Steps of Checkout
+-----------------
+
+Checkout in Sylius is divided into 4 steps. Each of these steps occurs when the Order goes into a certain state.
+See the Checkout state machine in the ``CoreBundle\Resources\config\app\state_machine.yml``
+together with the routing file for checkout: ``ShopBundle\Resources\config\routing\checkout.yml``.
+
+.. note::
+
+    Before performing Checkout :doc:`you need to have an Order created </book/orders>`.
+
+Addressing
+~~~~~~~~~~
+
+This is a step where the customer provides both **shipping and billing addresses**.
+
++--------------------------+----------------------------------------------------+
+| Transition after step    | Template                                           |
++--------------------------+----------------------------------------------------+
+| ``cart``-> ``addressed`` | ``SyliusShopBundle:Checkout:addressing.html.twig`` |
++--------------------------+----------------------------------------------------+
+
+How to perform the Addressing Step programmatically?
+''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Firstly if the **Customer** is not yet set on the Order it will be assigned depending on the case:
+
+* An already logged in **User** - the Customer is set for the Order using the ``Sylius\Bundle\CoreBundle\EventListener\CartBlamerListener``, that determines the user basing on the event.
+* An existent **User** that is not logged in - they are redirected to log in before continuing.
+* A **Customer** that was present in the system before (we've got their e-mail) - the Customer instance is updated via cascade, the order is assigned to it.
```

Comment:
It is unclear(at least to me), where did we store a customer. One was here before, but how can we determine it?